### PR TITLE
fix(anvil): purge old on disk state

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Blocked by https://github.com/foundry-rs/foundry/pull/4188

ref https://github.com/foundry-rs/foundry/issues/3623

this enforces a limit on the number of history snapshots we keep on disk. Prior to this, it was uncapped resulting in blow up mentioned in #3623

### CODE REVIEW

File: anvil/src/eth/backend/mem/storage.rs [hunk 1]
```diff
@@ -49,13 +63,31 @@ impl InMemoryBlockStates {
...
+    pub fn update_interval_mine_block_time(&mut self, block_time: Duration) {
+        self.min_in_memory_limit = INTERVAL_MINING_HISTORY_LIMIT;
+        let block_time = block_time.as_secs();
+        let max = DEFAULT_HISTORY_LIMIT as u64;
+        // a low block time would result in a lot of additional state (block)
+        let limit = (max.saturating_mul(block_time) / max) as usize;
+
+        self.in_memory_limit = limit.clamp(self.min_in_memory_limit, DEFAULT_HISTORY_LIMIT);
+
+        self.enforce_limits();
```
Comment (rkrasiuk) [inline anvil/src/eth/backend/mem/storage.rs:88]:
```suggestion
        let block_time = block_time.as_secs();
        let max = DEFAULT_HISTORY_LIMIT as u64;
        // a low block time would result in a lot of additional state (block)
        let limit = (max.saturating_mul(block_time) / max) as usize;

        self.in_memory_limit = limit.clamp(INTERVAL_MINING_HISTORY_LIMIT, DEFAULT_HISTORY_LIMIT);

        self.enforce_limits();
```

File: anvil/src/eth/backend/mem/storage.rs [hunk 2]
```diff
@@ -118,16 +167,16 @@ impl InMemoryBlockStates {
 impl fmt::Debug for InMemoryBlockStates {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("InMemoryBlockStates")
-            .field("limit", &self.limit)
+            .field("limit", &self.in_memory_limit)
```
Comment (rkrasiuk) [inline anvil/src/eth/backend/mem/storage.rs:170]:
```suggestion
            .field("in_memory_limit", &self.in_memory_limit)
            .field("max_on_disk_limit", &self.max_on_disk_limit)
```

**Review by rkrasiuk — APPROVED** [review]
lgtm, some nits

File: anvil/src/eth/backend/mem/storage.rs (hunk 1 above)
Comment (mattsse) [reply anvil/src/eth/backend/mem/storage.rs:88]: what's the difference here? hard to see in 8 loc

Comment (mattsse) [reply anvil/src/eth/backend/mem/storage.rs:88]: ah nvm, found it

Comment (mattsse) [reply anvil/src/eth/backend/mem/storage.rs:88]: but I disagree with this suggestion, since this ignore the currently configured limit

**Pull request merged**